### PR TITLE
#12274 - reshared entities archiving/dearchiving will trigger a Survn…

### LINKED ISSUE #12274: Reshared entities to Survnet (cases and events) will not trigger a Survnet call when archiving/dearchiving
### Bug Description

The entities (cases/events) which were shared with Survnet (in single or bulk mode), were soft-deleted, restored and then reshared with Survnet will not trigger a Survnet call when archiving/dearchiving which causes inconsistent data between Sormas entities and Survnet entities.

### Reproduction Steps

1. Share one or multiple entities with Survnet ( use Send to reporting tool for cases or events in single or in bulk mode)
2. Soft delete the shared entity/entities
3. Restore the entity/entities
4. Share again with Survnet the entity/entities
5. Try to archive/dearchive the entity/entities

The call to change the archive/dearchive flag in Survnet will not be made and therefore the report will not be downloaded in Survnet output folder.

### Expected Behavior

In the above mentioned scenario the call to Survnet in order to change the archive flag should be made and as an outcome the report should be downloaded.

The problem can be reproduced with any user with archive rights (Survnet project is needed to be installed).

### Affected Area(s)

Web app

### SORMAS Version

1.86

### Android version/Browser

Chrome

### User Role(s)

National user

### Server URL

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by carina29**
…et call

Fixes #12274

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12303

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12303

File: sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java [hunk 1]
```diff
@@ -1142,6 +1144,26 @@ public void setArchiveInExternalSurveillanceToolForEntities(List<String> entityU
 		}
 	}
 
+	public List<String> getSharedCaseUuids(List<String> entityUuids) {
+		List<Long> eventIds = getCaseIds(entityUuids);
```
Comment (MateStrysewske) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java:1148]:
```suggestion
		List<Long> caseIds = getCaseIds(entityUuids);
```

File: sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java [hunk 2]
```diff
@@ -1142,6 +1144,26 @@ public void setArchiveInExternalSurveillanceToolForEntities(List<String> entityU
...
+				sharedCaseUuids.add(shareInfo.getAssociatedObjectUuid());
+			}
+		});
+
+		return sharedCaseUuids;
+	}
+
+	public List<Long> getCaseIds(List<String> entityUuids) {
+		List<Long> caseIds = new ArrayList<>();
+		entityUuids.stream().forEach(uuid -> caseIds.add(this.getByUuid(uuid).getId()));
```
Comment (MateStrysewske) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java:1163]: `stream()` call is unnecessary here (same in EventService)

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12303

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**